docs/DEX.md

```diff
@@ -0,0 +1,115 @@
+
+Since `Schemas` and `Interfaces` are content-addressed, they are stored by the node, and not by specific communities. `Implementations` are community specific, because they reference paths to data held under specific communities.
+
+
+## The DEX Node API
+
+The DEX Node API is a read only API accessible over HTTPS through the reverse proxy at addresses following this pattern:
+
+```<node_host>:<reverse_proxy_port>/dex```
+
+### `GET /dex/<community_id>/<datastore_id>/<namespace>/<content_identifier>`
+
+
+### `GET /dex/schema/<schema_hash>`
+The schema hash is a hash of the schema data. Schemas are structued as JSON Schemas. 
+
+### `GET /dex/interface/<interface_hash>`
+```
+{
+    "schema_hash": "age4y43ysgsggh",
+    "description": "This is an interface for blog posts"
+}
+```
+
+### `GET /dex/<community_id>/implementation/<interface_hash>`
+```
+{
+    "interface_hash": "ageg32543ghdfhdfh",
+    "implementations": {
+        "datastore_1": "<content_identifier>",
+        "datastore_2": "<content_identifier>",
+        "datastore_3": "<content_identifier>",
+    }
+}
+```
+
+## The DEX proxy
+There are two layers of request forwarding in DEX. The first layer is handled explicitly by DEX, while the second is forwarding handled internally by datastores. The [network topology](src/TOPOLOGY.md) of the community can be used to determine proxying rules.
+
+### DEX Proxy Forwarding
+If the current node fielding a request does not host an instance of the datastore that is being requested, it can forward the request to the proper node.
+
+### Datastore Forwarding
+Internally, datastore instances that are part of the same process can forward requests between themselves, through a protocol of their own choosing. This will support DHTs, read replicas, and other forms of distributed data.
+
+
+## The DEX datastore driver API
+
+The driver API is very similare to the Node API. In fact, the for the `schema` and `interface` endpoints, DEX forwards the request to each datastore to see if one has the result. (Eh.. not sure about this, we might want to default to one, but everything should be able to implement it at least).
```

Comment:
i'm confused - is this the samea s getting schemas/interfaces from the DATA.md doc aboce? what is the DEX-level driver for?